Godeps/_workspace/src/github.com/GoogleCloudPlatform/kubernetes/pkg/client/transport.go

```diff
@@ -17,16 +17,174 @@ limitations under the License.
 package client
 
 import (
+	"bytes"
 	"crypto/tls"
 	"crypto/x509"
-	"encoding/base64"
 	"fmt"
 	"io/ioutil"
 	"net/http"
+	"time"
 
 	"github.com/golang/glog"
+
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/util"
+)
+
+type RequestInfo struct {
+	RequestHeaders http.Header
+	RequestVerb    string
+	RequestURL     string
+	RequestBody    []byte
+
+	ResponseStatus  string
+	ResponseBody    []byte
+	ResponseHeaders http.Header
+	ResponseErr     error
+
+	Duration time.Duration
+}
+
+func NewRequestInfo(req *http.Request, readBody bool) *RequestInfo {
+	reqInfo := &RequestInfo{}
+	reqInfo.RequestURL = req.URL.String()
+	reqInfo.RequestVerb = req.Method
+	reqInfo.RequestHeaders = req.Header
+	if readBody && req.Body != nil {
+		if body, err := ioutil.ReadAll(req.Body); err == nil {
+			reqInfo.RequestBody = body
+			req.Body = ioutil.NopCloser(bytes.NewBuffer(body))
+		}
+	}
+
+	return reqInfo
+}
+
+func (r *RequestInfo) Complete(response *http.Response, err error, readBody bool) {
+	if err != nil {
+		r.ResponseErr = err
+		return
+	}
+	r.ResponseStatus = response.Status
+	r.ResponseHeaders = response.Header
+	if readBody && response.Body != nil {
```

Comment:
@liggitt since you're here... Is there a way to figure out if this is really a stream?  I don't want to look through streams, but if you do it now, you never return.